File: backend/src/main/java/com/new_cafe/app/backend/entity/Menu.java
package com.new_cafe.app.backend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Menu {
    private Long id;
    private String korName;
    private String engName;
    private String description;
    private int price;
    private String image;
    private Long categoryId;
    private Boolean isAvailable;
    private Boolean isSoldOut;
    private Integer sortOrder;
    private java.sql.Timestamp createdAt;
    private java.sql.Timestamp updatedAt;

    private Category category;
}

File: frontend/types/menu.ts
// Menu Image Type
export interface MenuImage {
    id: string;
    url: string;
    isPrimary: boolean;
    sortOrder: number;
}

// Menu Option Item Type
export interface OptionItem {
    id: string;
    name: string;
    priceDelta: number; // 추가 가격 (0이면 무료)
}

// Menu Option Type
export interface MenuOption {
    id: string;
    name: string;
    type: 'radio' | 'checkbox';
    required: boolean;
    items: OptionItem[];
}

// Menu Category Type
export interface MenuCategory {
    id: string;
    korName: string;
    engName: string;
    icon?: string;
    sortOrder: number;
}

// Menu Type
export interface Menu {
    id: string;
    korName: string;
    engName: string;
    description: string;
    price: number;
    category: MenuCategory;
    images: MenuImage[];
    isAvailable: boolean;
    isSoldOut: boolean;
    sortOrder: number;
    options: MenuOption[];
    createdAt: Date;
    updatedAt: Date;
}

// Menu Status for filtering
export type MenuStatus = 'all' | 'available' | 'soldOut' | 'hidden';
